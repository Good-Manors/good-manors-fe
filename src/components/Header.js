import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.css';

export default function Header() {
  return (
    <header className={styles.Header}>
      <img src="src/assets/good-manors-logo.png" />
      <div id="app"></div>
      <nav>
        <Link to="/">Dashboard</Link>
        <Link to="/about-us">About Us</Link>
      </nav>
    </header >
  );
}