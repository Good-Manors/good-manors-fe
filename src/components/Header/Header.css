.Header {
  grid-area: header;
  background:url(../../assets/img-vertical/cat-yard-vert.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  padding-top: 2rem;
  background-color: black;
  transition: 3s;
}


  @media only screen and (max-width: 450px) {
    .Header{
      width: 100vw;
      padding: 1.5rem;
    }
  }

