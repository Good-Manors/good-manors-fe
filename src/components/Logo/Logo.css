.Logo {
  width: 10rem;
  height: 10rem;
  border-radius: 50%;
  background: white;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-content: center;

  img {
    display: block;
    margin: auto auto;
    width: 80%;
  }
}
