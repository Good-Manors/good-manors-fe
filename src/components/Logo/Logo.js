import React from 'react';
import styles from './Logo.css';
import logo from '../../assets/good-manors-logo.png';

const Logo = () => {

  return (
    <div id='logo-background' className={styles.Logo}>
      <img src={logo} />
    </div>
  );
};


export default Logo;
