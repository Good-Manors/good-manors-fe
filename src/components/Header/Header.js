import React, { useState, useEffect, useRef } from 'react';
import bathroom from '../../assets/img-vertical/bathroom-vert.jpg';
import catYard from '../../assets/img-vertical/cat-yard-vert.jpg';
import dining from '../../assets/img-vertical/dining-vert.jpg';
import garden from '../../assets/img-vertical/garden-vert.jpg';
import kidRoom from '../../assets/img-vertical/kid-room-vert.jpg';
import kitchen from '../../assets/img-vertical/kitchen-vert.jpg';
import nook from '../../assets/img-vertical/nook-vert.jpg';
import styles from './Header.css';
import Logo from '../Logo/Logo';

const bgImageArrayVert = [bathroom, dining, garden, kidRoom, kitchen, nook, catYard];

export default function Header() {
  const bgRef = useRef(null);

  const [bg1, setBg1] = useState(0);
  const [bg2, setBg2] = useState(1);
  const [bgCurrent, setBgCurrent] = useState('bg1');

  useEffect(() => {
    const bgMap = {
      bg1,
      bg2
    };

    setTimeout(() => {
      if(!bgRef.current) return;
      bgRef.current.style.background = `url(${bgImageArrayVert[bgMap[bgCurrent]]}) no-repeat center`;
      bgRef.current.style.backgroundSize = 'cover';

      if(bgCurrent === 'bg1') {
        setBgCurrent('bg2');
        if(bg1 + 2 >= bgImageArrayVert.length && bg2 === 0) {
          setBg1(1);
        } else if(bg1 + 2 >= bgImageArrayVert.length) {
          setBg1(0);
        } else {
          setBg1(bg1 + 2);
        }
      }

      if(bgCurrent === 'bg2') {
        setBgCurrent('bg1');
        if(bg2 + 2 >= bgImageArrayVert.length && bg1 === 0) {
          setBg2(1);
        } else if(bg2 + 2 >= bgImageArrayVert.length) {
          setBg2(0);
        } else {
          setBg2(bg2 + 2);
        }
      }
    }, 6000);
  }, [bgCurrent]);

  return (
    <header ref={bgRef} className={styles.Header}>
      <Logo />
    </header >
  );
}
