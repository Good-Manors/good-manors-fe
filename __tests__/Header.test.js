import React from 'react';
import { shallow } from 'enzyme';
import Header from '../src/components/Header/Header';
import Logo from '../src/components/Logo/Logo';


describe('Header and Logo', () => {
  
  it('Should render a Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Logo', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
