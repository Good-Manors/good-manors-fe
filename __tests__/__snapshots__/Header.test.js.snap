// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header and Logo Logo 1`] = `
<div
  className="Logo"
  id="logo-background"
>
  <img
    src="src/assets/good-manors-logo.png"
  />
</div>
`;

exports[`Header and Logo Should render a Header 1`] = `
<header
  className="Header"
>
  <Logo />
</header>
`;
